src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -290,7 +290,7 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                             });
                     }
                     
-                    if (!(result.IRContext.ResultType == node.IRContext.ResultType || result is ErrorValue || result.IRContext.ResultType is BlankType || func is IfErrorFunction))
+                    if (!(result.IRContext.ResultType._type == node.IRContext.ResultType._type || result is ErrorValue || result.IRContext.ResultType is BlankType || func is IfErrorFunction))

```

Comment:
>_type [](http://example.com/codeflow?start=54&length=5)

Do we need this? Looks like it's to invoke Dtype's equality. 
FormulaType overrides == already and forward to Dtype?

Reply:
Yes, we should only rely on DType inside Binder...

Think of 2 different implementation of RecordType, If they have same underlying dtype, should this check pass?